src/gui/elevation/qgselevationcontrollerwidget.cpp

```diff
@@ -0,0 +1,290 @@
+{
+  if ( range == mCurrentRange )
+    return;
+
+  mCurrentRange = range;
+  mBlockSliderChanges = true;
+  mSlider->setRange( static_cast< int >( std::floor( range.lower() * mSliderPrecision ) ),
+                     static_cast< int >( std::ceil( range.upper() * mSliderPrecision ) ) );
+  mBlockSliderChanges = false;
+  emit rangeChanged( range );
+
+  mSliderLabels->setRange( mCurrentRange );
+}
+
+void QgsElevationControllerWidget::setRangeLimits( const QgsDoubleRange &limits )
+{
+  if ( limits.isInfinite() )
+    return;
+
+  mRangeLimits = limits;
+
+  const double limitRange = limits.upper() - limits.lower();
+
+  // pick a reasonable slider precision, given that the slider operates in integer values only
+  mSliderPrecision = std::max( 1000, mSlider->height() ) / limitRange;
+
+  mBlockSliderChanges = true;
+  mSlider->setRangeLimits( static_cast< int >( std::floor( limits.lower() * mSliderPrecision ) ),
+                           static_cast< int >( std::ceil( limits.upper() * mSliderPrecision ) ) );
+
+  // clip current range to fit limits
+  const double newCurrentLower = std::max( mCurrentRange.lower(), limits.lower() );
+  const double newCurrentUpper = std::min( mCurrentRange.upper(), limits.upper() );
+  const bool rangeHasChanged = newCurrentLower != mCurrentRange.lower() || newCurrentUpper != mCurrentRange.upper();
+
+  mSlider->setRange( static_cast< int >( std::floor( newCurrentLower * mSliderPrecision ) ),
+                     static_cast< int >( std::ceil( newCurrentUpper * mSliderPrecision ) ) );
+  mCurrentRange = QgsDoubleRange( newCurrentLower, newCurrentUpper );
+  mBlockSliderChanges = false;
+  if ( rangeHasChanged )
+    emit rangeChanged( mCurrentRange );
+
+  mSliderLabels->setLimits( mRangeLimits );
+}
+
+void QgsElevationControllerWidget::updateWidgetMask()
+{
+  // we want mouse events from this widgets children to be caught, but events
+  // on the widget itself to be ignored and passed to underlying widgets which are NOT THE DIRECT
+  // PARENT of this widget.
+  // this is definitively *****NOT***** possible with event filters, by overriding mouse events, or
+  // with the WA_TransparentForMouseEvents attribute
+
+  QRegion reg( frameGeometry() );
+  reg -= QRegion( geometry() );
+  reg += childrenRegion();
+  setMask( reg );
+}
+
+//
+// QgsElevationControllerLabels
+//
+///@cond PRIVATE
+QgsElevationControllerLabels::QgsElevationControllerLabels( QWidget *parent )
+  : QWidget( parent )
+{
+  const QFontMetrics fm( font() );
+  setMinimumWidth( fm.horizontalAdvance( '0' ) * 5 );
+  setAttribute( Qt::WA_TransparentForMouseEvents );
+}
+
+void QgsElevationControllerLabels::paintEvent( QPaintEvent * )
+{
+  QStyleOptionSlider styleOption;
+  styleOption.initFrom( this );
+
+  const QRect sliderRect = style()->subControlRect( QStyle::CC_Slider, &styleOption,  QStyle::SC_SliderHandle, this );
+  const int sliderHeight = sliderRect.height();
+
+  QFont f = font();
```

Comment:
Wondering if we should go for a smaller font than the default one, maybe match the size with that of the status bar?

```
  // Drop the font size in the status bar by a couple of points
  QFont statusBarFont = font();
  int fontSize = statusBarFont.pointSize();
#ifdef Q_OS_WIN
  fontSize = std::max( fontSize - 1, 8 ); // bit less on windows, due to poor rendering of small point sizes
#else
  fontSize = std::max( fontSize - 2, 6 );
#endif
  statusBarFont.setPointSize( fontSize );
  statusBar()->setFont( statusBarFont );
```

Reply:
Ok ,done!

Reply:
@nyalldawson , thanks